src/synthesizer/photometry.py

```diff
@@ -0,0 +1,334 @@
+
+    def __getitem__(self, filter_code):
+        """
+        Enable dictionary key look up syntax to extract specific photometry,
+        e.g. Sed.broadband_luminosities["JWST/NIRCam.F150W"].
+
+        NOTE: this will always return photometry with units. Unitless
+        photometry is accessible in array form via self._rest_photometry
+        or self._obs_photometry based on what frame is desired. For
+        internal use this should be fine and the UI (where this method
+        would be used) should always return with units.
+
+        Args:
+            filter_code (str)
+                The filter code of the desired photometry.
+        """
+
+        # Perform the look up
+        return self._look_up[filter_code]
+
+    def keys(self):
+        """
+        Enable dict.keys() behaviour.
+        """
+        return self._look_up.keys()
+
+    def values(self):
+        """
+        Enable dict.values() behaviour.
+        """
+        return self._look_up.values()
+
+    def items(self):
+        """
+        Enables dict.items() behaviour.
+        """
+        return self._look_up.items()
+
+    def __iter__(self):
+        """
+        Enable dict iter behaviour.
+        """
+        return iter(self._look_up.items())
+
+    def __str__(self):
+        """
+        Allow for a summary to be printed.
+
+        Returns:
+            str: A formatted string representation of the PhotometryCollection.
+        """
+
+        # Define the headers
+        headers = [
+            f"{f.filter_code} (\u03BB = {f.pivwv().value:.2e} {str(f.lam.units)})"
+            for f in self.filters
+        ]
+
+        # Determine the width of each column
+        widths = [
+            max(
+                len(header),
+                len(str(format(phot.value, ".2e")) + " " + str(phot.units)),
+            )
+            + 2
+            for header, phot in zip(headers, self._look_up.values())
+        ]
+
+        # How many characters make up the table?
+        tot_width = sum(widths) + len(widths) + 1
+
+        # Create the table header
+        head = "|".join(
+            f"{header.center(width)}" for header, width in zip(headers, widths)
+        )
+        sep = "|".join("-" * width for width in widths)
+
+        # Create the photometry row
+        phot = "|".join(
+            f"{str(format(self[key].value, '.2e'))} "
+            f"{str(self[key].units)}".center(width)
+            for key, width in zip(self.filter_codes, widths)
+        )
+
+        # Create the centered title with "=" on either side
+        if self.rest_frame:
+            title = f"{'= REST FRAME PHOTOMETRY ='.center(tot_width, '=')}"
+        else:
+            title = f"{'= OBSERVED PHOTOMETRY ='.center(tot_width, '=')}"
+
+        # Combine everything into the final table
+        table = (
+            f"{title}\n-{sep.replace('|', '-')}-\n|{head}|"
+            f"\n|{sep}|\n|{phot}|\n-{sep.replace('|', '-')}-"
+        )
+
+        return table
```

Comment:
maybe I'm misunderstanding how this works, but surely a two column table, with name and details, would be much easier to read, especially where there are >10 filters in a collection?

Reply:
Done, much better!